### PR TITLE
make images configurable in track/0.15

### PR DESCRIPTION
**Pull request by NohaIhab**
part of #144 

## Summary of changes:
1. Cherry-picking commits to ensure images are configurable for airgapped environment.

  Only the following commits were cherry-picked:
  - 9bfa7a2
  - ea9dcfc
  - 10f75f9
  
2. modified the json file to have images versions `v0.15.0`

## Testing get-images script

```
bash tools/get-images.sh
08:07:36 initCommand [WARN] JSON file output is now JSON by default (instead of yaml). Use '-oy' or '--output-format=yaml' for yaml output
docker.io/kubeflowkatib/katib-db-manager:v0.15.0
docker.io/kubeflowkatib/katib-controller:v0.15.0
docker.io/kubeflowkatib/katib-ui:v0.15.0
docker.io/kubeflowkatib/mxnet-mnist:v0.15.0
docker.io/kubeflowkatib/earlystopping-medianstop:v0.15.0
docker.io/kubeflowkatib/enas-cnn-cifar10-cpu:v0.15.0
docker.io/kubeflowkatib/file-metrics-collector:v0.15.0
docker.io/kubeflowkatib/file-metrics-collector:v0.15.0
docker.io/kubeflowkatib/tfevent-metrics-collector:v0.15.0
docker.io/kubeflowkatib/pytorch-mnist-cpu:v0.15.0
docker.io/kubeflowkatib/pytorch-mnist-cpu:v0.15.0
docker.io/kubeflowkatib/suggestion-hyperopt:v0.15.0
docker.io/kubeflowkatib/suggestion-hyperopt:v0.15.0
docker.io/kubeflowkatib/suggestion-optuna:v0.15.0
docker.io/kubeflowkatib/suggestion-hyperband:v0.15.0
docker.io/kubeflowkatib/suggestion-skopt:v0.15.0
docker.io/kubeflowkatib/suggestion-goptuna:v0.15.0
docker.io/kubeflowkatib/suggestion-goptuna:v0.15.0
docker.io/kubeflowkatib/suggestion-optuna:v0.15.0
docker.io/kubeflowkatib/suggestion-enas:v0.15.0
docker.io/kubeflowkatib/suggestion-darts:v0.15.0
docker.io/kubeflowkatib/suggestion-pbt:v0.15.0
```

### CODE REVIEW

**Comment by i-chvets** [comment]
There are many changes to `requirements.*` files. Even for charms that do not have images related changes. Can we verify that these changes are really necessary?

File: charms/katib-controller/requirements-integration.in [hunk 1]
```diff
@@ -0,0 +1,3 @@
+juju==2.9.44
```
Comment (DnPlas) [inline charms/katib-controller/requirements-integration.in:1]: I'd leave this as 

```
# Pinning to <3.0 due to compatibility with the 2.9 controller version and 3.0 not being maintained anymore
juju<3.0
```

**Review by DnPlas — COMMENTED** [review]
Thanks @NohaIhab. Small comment.

Regarding the requirement files, almost all make sense as we are introducing new integration tests for katib-controller as well as pinning pyyaml to 6.0.1 because of that issue. I don't think the issue is present anymore, but we can leave it as is to keep consistency with `main`.

**Review by DnPlas — APPROVED** [review]

**Comment by i-chvets** [comment]
> Regarding the requirement files, almost all make sense as we are introducing new integration tests for katib-controller as well as pinning pyyaml to 6.0.1 because of that issue. I don't think the issue is present anymore, but we can leave it as is to keep consistency with `main`.

Just to note here, that we are removing pinning of pyyaml in[ other PRs](https://github.com/canonical/kserve-operators/pull/180) for this effort.

**Comment by orfeas-k** [comment]
Thank you @i-chvets for pointing this out. I think the rationale behind keeping it here is that it is pinned also in `main` branch in `katib-operators` while other repositories may have follow up PRs that removed it.

**Pull request merged**